test/hotspot/jtreg/compiler/lib/template_framework/Renderer.java

```diff
@@ -343,47 +353,79 @@ private void renderStringWithDollarAndHashtagReplacements(String s) {
             // Find the next "$" or "#", after start.
             int dollar  = s.indexOf("$", start);
             int hashtag = s.indexOf("#", start);
+            // If the character was not found, we want to have the rest of the
+            // String s, so instead of "-1" take the end/length of the String.
             dollar  = (dollar == -1)  ? s.length() : dollar;
```

Comment:
`s.length()` could be called once before the loop and then reused inside the loop.